File: 0945_minimum_increment_to_make_array_unique/solution.cpp
// LeetCode 0945 - Minimum Increment to Make Array Unique
// https://leetcode.com/problems/minimum-increment-to-make-array-unique/

#include <algorithm>
#include <vector>

class Solution {
public:
    int minIncrementForUnique(std::vector<int>& nums) {
        std::sort(nums.begin(), nums.end());
        int ans = 0;
        for (int i = 1; i < (int)nums.size(); i++) {
            if (nums[i] <= nums[i - 1]) {
                int need = nums[i - 1] + 1;
                ans += need - nums[i];
                nums[i] = need;
            }
        }
        return ans;
    }
};

File: 0945_minimum_increment_to_make_array_unique/Solution.kt
// LeetCode 0945 - Minimum Increment to Make Array Unique
// https://leetcode.com/problems/minimum-increment-to-make-array-unique/

class Solution {
    fun minIncrementForUnique(nums: IntArray): Int {
        nums.sort()
        var ans = 0
        for (i in 1 until nums.size) {
            if (nums[i] <= nums[i - 1]) {
                var need = nums[i - 1] + 1
                ans += need - nums[i]
                nums[i] = need
            }
        }
        return ans
    }
}
